crates/router/src/db/merchant_connector_account.rs

```diff
@@ -467,9 +470,16 @@ impl MerchantConnectorAccountInterface for Store {
                 "profile_id".to_string(),
             ))?;
 
-            super::cache::publish_and_redact(
+            super::cache::publish_and_redact_multiple(
                 self,
-                cache::CacheKind::Accounts(format!("{}_{}", mca.merchant_id, _profile_id).into()),
+                [
+                    cache::CacheKind::Accounts(
+                        format!("{}_{}", mca.merchant_id, _profile_id).into(),
+                    ),
+                    cache::CacheKind::KGraph(
+                        format!("kgraph_{}_{_profile_id}", mca.merchant_id).into(),
```

Comment:
Feature gating for `business_profile_routing`

Reply:
we have a check for profile_id here, we seem to be throwing an error in case profile_id is not present.